keymanager/src/crypto/kdf.rs

```diff
@@ -687,146 +757,203 @@ mod tests {
-        let mut req3 = req1.clone();
-        let mut req4 = req1.clone();
-        req2.runtime_id = Namespace::from(vec![2u8; 32]);
-        req3.key_pair_id = KeyPairId::from(vec![3u8; 32]);
-        req4.epoch = 2;
-
-        let mut seeds = HashSet::new();
-        seeds.insert(req1.seed());
-        seeds.insert(req2.seed());
-        seeds.insert(req3.seed());
-        seeds.insert(req4.seed());
-
-        assert_eq!(seeds.len(), 4);
+    fn unique_customs() {
+        // All key requests must have unique customs otherwise they can
+        // derivate keys from the same KMac and/or CShake!!!
+        let mut customs: HashSet<&[u8]> = HashSet::new();
+        customs.insert(&CHECKSUM_CUSTOM);
+        customs.insert(&RUNTIME_KDF_CUSTOM);
+        customs.insert(&RUNTIME_XOF_CUSTOM);
+        customs.insert(&EPHEMERAL_KDF_CUSTOM);
+        customs.insert(&EPHEMERAL_XOF_CUSTOM);
+        customs.insert(&RUNTIME_CHECKSUM_CUSTOM);
+        customs.insert(&RUNTIME_SIGNING_KEY_CUSTOM);
+        assert_eq!(7, customs.len());
     }
 
     #[test]
     fn vector_test() {
+        #[derive(Default)]
         struct TestVector<'a> {
             master_secret: &'a str,
-            seed: &'a str,
-            kdf_custom: &'a [u8],
-            xof_custom: &'a [u8],
-            sk: &'a str,
-            pk: &'a str,
+            ephemeral_secret: &'a str,
+            runtime_id: &'a str,
+            key_pair_id: &'a str,
+            epoch: EpochTime,
+            lk_sk: &'a str,
+            lk_pk: &'a str,
+            ek_sk: &'a str,
+            ek_pk: &'a str,
         }
 
-        let kdf_custom = b"54761365b5a007c2bd13cf880a8a58263f56bb7057f7548b7467e871f6f4bb1f";
-        let xof_custom = b"a0caee5ecac1a287f06ff8748cf37e809f049cf6866d668ecdbed3ce05b40f2c";
-
         let vectors = vec![
             // A random vector.
             TestVector {
                 master_secret: "3fa39161fe106fb770503558dad41bea3221888c09477d864507e615094f5d38",
-                seed: "2a9d51ed0380d11b26255ecc894dfc3aa48196c9b0da463a3073ea820a65a090",
-                kdf_custom: &RUNTIME_KDF_CUSTOM,
-                xof_custom: &RUNTIME_XOF_CUSTOM,
-                sk: "c818f97228a53b201e2afa383b64199c732c7ba1a67ac55ea2cb3b8fba367740",
-                pk: "fb99076f6a5a8c52bcf562cae5152d0410f47a615fc7bc4966dc9a4f477bd217",
+                ephemeral_secret:
+                    "2a9d51ed0380d11b26255ecc894dfc3aa48196c9b0da463a3073ea820a65a090",
+                runtime_id: "c818f97228a53b201e2afa383b64199c732c7ba1a67ac55ea2cb3b8fba367740",
+                key_pair_id: "fb99076f6a5a8c52bcf562cae5152d0410f47a615fc7bc4966dc9a4f477bd217",
+                epoch: 8935419451,
+                lk_sk: "702c5a25ff1149591cf7bca763fbe647a43939f43330a8fd331305d73e489b7c",
+                lk_pk: "66013b5ad263b4fa504e8682371fbfa102d18cc1e1b811bc125603b1abc9487b",
+                // lk_sk: "a009036a2a796c3bf1ae498e7055b481bc5965f9ff013c25270b1a61c8da125d",
```

Comment:
What are these commented out values?

Reply:
Those are correct values once we fix problem mentioned above (KDF - XOF).

Reply:
Ah ok, can you add that to the comment?